sale_partner_selectable_option_extended/models/res_partner.py

```diff
@@ -0,0 +1,30 @@
+# © 2025 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    @api.model_create_multi
+    def create(self,vals):
+        partners = super(ResPartner, self).create(vals)
+        for partner in partners:
+            if partner.type != 'contact':
+                partner.sale_selectable = partner.parent_id.sale_selectable
+            else:
+                partner.sale_selectable = (
+                self.env["ir.config_parameter"]
+                .sudo()
+                .get_param("sale_partner_selectable_option.default_sale_selectable")
+            )
+        return partner
+
+    @api.onchange("sale_selectable")
+    def onchange_sale_selectable_parent(self):
+        for partner in self:
+            if partner.child_ids:
+                for child in partner.child_ids:
+                    if child.type != "contact":
+                        child.sale_selectable = partner.sale_selectable
```

Comment:
Try to simplify this if -> for -> if nested statements in a single operation using `filtered` method.